extensions/spring-data-jpa/runtime/src/main/java/io/quarkus/spring/data/runtime/RepositorySupport.java

```diff
@@ -24,13 +24,21 @@ public static List<?> findByIds(AbstractJpaOperations<PanacheQuery<?>> operation
                 result.add(byId);
             }
         }
-        return result;
+        return operations.findByIds(entityClass, result);
```

Comment:
Are you sure about this one? It appears we do `findById` on each ID, and then call `findByIds` with the results (which should be a list of entities, not ids).